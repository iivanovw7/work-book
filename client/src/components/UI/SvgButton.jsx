import PropTypes from 'prop-types';
import React from 'react';
import styled, { ThemeProvider } from 'styled-components';
import { ripples } from '../../styles';
import { btnBackground, btnLighten } from '../../theme';
/* eslint react/require-default-props: 0 */

const StyledIconButton = styled.button`
	${ripples};
  display: flex;
  flex-wrap: nowrap;
  padding: 0.3rem;
  text-align: center;
  flex-direction: row;
  color: ${btnBackground};
  align-items: center;
  border: none;
  background: none;
  outline: none;
  
  &:hover {
    color: ${btnLighten};
    transition: all 0.2s ease-in-out;
    user-select: none;
    cursor: pointer;
  }
  
  &:focus {
    outline: none;
  }
  
  &::-moz-focus-inner {
    border: 0;
  }

`;

const SvgButton = (props) => {
  const {
    Image, variant, handleClick, text, width, height
  } = props;

  return (
    <ThemeProvider theme={{ mode: 'dark' }}>
      <StyledIconButton variant={variant} onClick={handleClick}>
        <Image width={width} height={height} alt={text} />
      </StyledIconButton>
    </ThemeProvider>
  );
};

StyledIconButton.propTypes = {
  variant: PropTypes.oneOf(['primary', 'secondary'])
};

StyledIconButton.defaultProps = {
  variant: 'primary'
};

SvgButton.propTypes = {
  handleClick: PropTypes.func,
  Image: PropTypes.any.isRequired,
  variant: PropTypes.string.isRequired,
  height: PropTypes.string.isRequired,
  width: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired
};

export default SvgButton;
