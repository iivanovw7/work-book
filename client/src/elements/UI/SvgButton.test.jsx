import 'cross-fetch/polyfill';
import 'jsdom-global/register';
import React from 'react';
import { mount } from 'enzyme';
import SvgButton from './SvgButton';

// In order to make webpack svgr loader with jest
// check /__mocks__/svgrMock.js and /jest.config.js
import { ReactComponent as IconEng } from '../../assets/img/eng.svg';

jest.mock('../../config/apiURL', () => () => '/api');

describe('Testing SVG Button component', () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  const props = {
    Image: IconEng,
    variant: 'primary',
    height: '10',
    width: '10',
    text: 'text'
  };

  const Composition = () => (
    <SvgButton userAccess {...props} />
  );

  it('Should match snapshot', () => {
    const component = mount(<Composition />);

    expect(component).toMatchSnapshot();
  });

  it('Should render with all child elements', () => {
    const component = mount(<Composition />);

    expect(component.find('icon-mock')).toHaveLength(1);
    expect(component.find('icon-mock').prop('alt')).toBe('text');
  });

  it('Should render with correct props', () => {
    const component = mount(<Composition />);

    expect(component.find(SvgButton).prop('height')).toBe('10');
    expect(component.find(SvgButton).prop('width')).toBe('10');
  });

  it('Should trigger click event', () => {
    const mockFunc = jest.fn();
    const component = mount(<Composition handleClick={mockFunc()} />);

    component.find(SvgButton).at(0).simulate('click');
    expect(component.find(SvgButton)).toHaveLength(1);
    expect(mockFunc).toHaveBeenCalled();
  });
});
