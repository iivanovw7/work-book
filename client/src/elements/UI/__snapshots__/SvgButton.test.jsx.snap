// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SVG Button component Should match snapshot 1`] = `ReactWrapper {}`;
